### PR TITLE
base torch image option on build

### LINKED ISSUE #52: [FEA] Add PyTorch base image option to container build
The container only builds on `nvcr.io/nvidia/cuda` today (Dockerfile L26,
`DAQIRI_BASE_TARGET` L18). Downstream users who want to land packets on the GPU
and run Torch / TensorRT inference in the same process have to layer Torch on
top of the DAQIRI image themselves.

Suggested fixes:
1. Keep the Dockerfile as-is and ship a small downstream `Dockerfile.torch`
   under `examples/` that `FROM`s the DAQIRI runtime image and adds Torch.
2. Add a `torch` option to `DAQIRI_BASE_TARGET` based on the NGC PyTorch image,
   alongside the existing `dpdk` / `rdma` / `gpunetio` / `rivermax` targets.

Ways to validate feature addition:
Validate with `BASE_TARGET=<choice> scripts/build-container.sh`,
`daqiri_bench_raw_sw_loopback.yaml`, and
`python3 -c "import torch; print(torch.cuda.is_available())"`.

### PR DESCRIPTION
**Pull request by chloecrozier**
## Summary

Adds a `BASE_IMAGE` env var to `scripts/build-container.sh` accepting
`cuda` (default, unchanged behavior) or `torch`. Setting `BASE_IMAGE=torch`
swaps the multi-stage Dockerfile's base image so the resulting container
ships Torch + Torch-TensorRT alongside DAQIRI, for downstream workflows
that ingest packets directly into GPU memory and run inference in the
same process.

| Knob | Resolves to |
|---|---|
| `BASE_IMAGE=cuda` (default) | `nvcr.io/nvidia/cuda:13.1.0-devel-ubuntu24.04` |
| `BASE_IMAGE=torch` | `nvcr.io/nvidia/pytorch:26.01-py3` |
| `BASE_IMAGE=<anything else>` | hard error, exits 1 |

## Why each Dockerfile change was needed

The PyTorch base does not ship the same pre-installed packages and apt
sources as the CUDA base. Three small changes make the Dockerfile work
on both bases instead of relying on what the CUDA image happens to
include:
1. **DOCA repository configured earlier.** The repo that provides the
   Mellanox networking libraries is now set up in the shared dependency
   stage instead of the DPDK stage. Same repo, same packages, just
   reachable from one stage higher.
2. **`mft` install moved.** `mft` is not in Ubuntu's default repos. It
   used to install in the very first stage and only succeeded because
   the CUDA image preconfigures NVIDIA's apt sources. It now installs
   in the same step as `mlnx-ofed-kernel-utils`, from the DOCA repo
   above, which works on any base.
3. **PyTorch tag pinned to `26.01-py3`.** This is the first NGC PyTorch
   release with CUDA 13, which the repo's `sm_121` (GB10 / DGX Spark)
   build target requires. Older tags fail to compile the GPUDirect
   benches on Blackwell hardware.

## Test plan

| # | What | Result |
|---|---|---|
| 1 | `BASE_IMAGE=cuda scripts/build-container.sh` (default path) | green |
| 2 | `IMAGE_TAG=daqiri-torch:local BASE_IMAGE=torch BASE_TARGET=dpdk DAQIRI_MGR="dpdk rdma socket" scripts/build-container.sh` | green |
| 3 | `docker run --rm --gpus all daqiri-torch:local python3 -c "import torch; print(torch.__version__, torch.cuda.is_available()); print(torch.cuda.get_device_name(0))"` on a GB10 host | `2.10.0a0+a36e1d39eb.nv26.01.42222806 True / NVIDIA GB10` |
| 4 | `daqiri_bench_socket` UDP loopback inside `daqiri-torch:local` (5s) | `tx_pkts=2000 rx_pkts=1984`, `1000/992` each direction |
| 5 | Same bench inside `daqiri-cuda:local` (regression) | identical numbers |
| 6 | `daqiri_bench_raw_gpudirect sw_loopback` inside `daqiri-torch:local` (5s) | 215,429,120 pkts TX = 215,429,120 pkts RX, zero loss, ~366 Gbps |
| 7 | `BASE_IMAGE=bogus scripts/build-container.sh` | `ERROR: invalid BASE_IMAGE='bogus'. Choose from: cuda, torch`, exit 1, no docker invocation |

## Doc impact

Per `.claude/rules/docs-sync.md`, the user-facing build command grew a
new option, so `README.md`, `docs/getting-started.md`, and
`docs/index.html` each get a one-line mention of `BASE_IMAGE=torch`.

Closes #52

### CODE REVIEW

File: scripts/build-container.sh [hunk 1]
```diff
@@ -3,13 +3,28 @@ set -euo pipefail
...
 DAQIRI_MGR="${DAQIRI_MGR:-dpdk socket}"
 DAQIRI_BUILD_PYTHON="${DAQIRI_BUILD_PYTHON:-OFF}"
 BUILD_SHARED_LIBS="${BUILD_SHARED_LIBS:-ON}"
 
+case "${BASE_IMAGE}" in
+  cuda)
+    DAQIRI_OS_BASE_IMAGE="nvcr.io/nvidia/cuda:13.1.0-devel-ubuntu24.04"
+    ;;
+  torch)
+    DAQIRI_OS_BASE_IMAGE="nvcr.io/nvidia/pytorch:26.01-py3"
```
Comment (cliffburdick) [inline scripts/build-container.sh:16]: Does pytorch include CUDA? If so, what version?

**Review by cliffburdick — APPROVED** [review]

**Review by RamyaGuru — APPROVED** [review]
Looks good! Your test plan covered everything I wanted to try.

File: scripts/build-container.sh (hunk 1 above)
Comment (chloecrozier) [reply scripts/build-container.sh:16]: Looks like `Build cuda_13.1.r13.1/compiler.37061995_0`
```
(base) ccrozier@spark-960b:/srv/nfs/share/users/ccrozier/daqiri$ docker run --rm nvcr.io/nvidia/pytorch:26.01-py3 \
  bash -c 'nvcc --version | tail -2; python3 -c "import torch; print(torch.version.cuda)"'

=============
== PyTorch ==
=============

NVIDIA Release 26.01 (build 256811084)
PyTorch Version 2.10.0a0+a36e1d3
Container image Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
Copyright (c) 2014-2024 Facebook Inc.
Copyright (c) 2011-2014 Idiap Research Institute (Ronan Collobert)
Copyright (c) 2012-2014 Deepmind Technologies    (Koray Kavukcuoglu)
Copyright (c) 2011-2012 NEC Laboratories America (Koray Kavukcuoglu)
Copyright (c) 2011-2013 NYU                      (Clement Farabet)
Copyright (c) 2006-2010 NEC Laboratories America (Ronan Collobert, Leon Bottou, Iain Melvin, Jason Weston)
Copyright (c) 2006      Idiap Research Institute (Samy Bengio)
Copyright (c) 2001-2004 Idiap Research Institute (Ronan Collobert, Samy Bengio, Johnny Mariethoz)
Copyright (c) 2015      Google Inc.
Copyright (c) 2015      Yangqing Jia
Copyright (c) 2013-2016 The Caffe contributors
All rights reserved.

Various files include modifications (c) NVIDIA CORPORATION & AFFILIATES.  All rights reserved.

GOVERNING TERMS: The software and materials are governed by the NVIDIA Software License Agreement
(found at https://www.nvidia.com/en-us/agreements/enterprise-software/nvidia-software-license-agreement/)
and the Product-Specific Terms for NVIDIA AI Products
(found at https://www.nvidia.com/en-us/agreements/enterprise-software/product-specific-terms-for-ai-products/).

NOTE: CUDA Forward Compatibility mode ENABLED.
  Using CUDA 13.1 driver version 590.48.01 with kernel driver version 580.95.05.
  See https://docs.nvidia.com/deploy/cuda-compatibility/ for details.

NOTE: The SHMEM allocation limit is set to the default of 64MB.  This may be
   insufficient for PyTorch.  NVIDIA recommends the use of the following flags:
   docker run --gpus all --ipc=host --ulimit memlock=-1 --ulimit stack=67108864 ...

Cuda compilation tools, release 13.1, V13.1.115
Build cuda_13.1.r13.1/compiler.37061995_0
13.1
```
